dialogue-core/src/main/java/com/palantir/dialogue/core/PinUntilErrorChannel.java

```diff
@@ -82,27 +82,41 @@ static PinUntilErrorChannel of(
             List<LimitedChannel> channels,
             DialoguePinuntilerrorMetrics metrics,
             Random random) {
+        return from(channels.get(0), strategy, channels, metrics, random);
+    }
+
+    static PinUntilErrorChannel from(
+            LimitedChannel initialChannel,
+            NodeSelectionStrategy strategy,
+            List<LimitedChannel> channels,
+            DialoguePinuntilerrorMetrics metrics,
+            Random random) {
         /**
          * The *initial* list is shuffled to ensure that clients across the fleet don't all traverse the in the
          * same order.  If they did, then restarting one upstream node n would shift all its traffic (from all
          * servers) to upstream n+1. When n+1 restarts, it would all shift to n+2. This results in the disastrous
          * situation where there might be many nodes but all clients have decided to hammer one of them.
          */
         ImmutableList<LimitedChannel> initialShuffle = shuffleImmutableList(channels, random);
+        int initialHost = initialShuffle.indexOf(initialChannel);
```

Comment:
We only rely on reference equality since in LimitedChannels are reused across updates